ComputeExamples.md

```diff
@@ -0,0 +1,676 @@
+<!--
+    assert(computeEnv, 'Cannot find the free compute env')
+-->
+
+Let's have 5 minute of compute access
+```Typescript
+    const mytime = new Date()
+    const computeMinutes = 5
+    mytime.setMinutes(mytime.getMinutes() + computeMinutes)
+    const computeValidUntil = Math.floor(mytime.getTime() / 1000)
+
+    const assets: ComputeAsset[] = [
+      {
+        documentId: resolvedDatasetDdo.id,
+        serviceId: resolvedDatasetDdo.services[0].id
+      }
+    ]
+    const dtAddressArray = [resolvedDatasetDdo.services[0].datatokenAddress]
+    const algo: ComputeAlgorithm = {
+      documentId: resolvedAlgorithmDdo.id,
+      serviceId: resolvedAlgorithmDdo.services[0].id
+    }
+
+    const providerInitializeComputeResults = await ProviderInstance.initializeCompute(
+      assets,
+      algo,
+      computeEnv.id,
+      computeValidUntil,
+      providerUrl,
+      consumerAccount
+    )
+```
+<!--
+    assert(!('error' in providerInitializeComputeResults), 'Cannot order algorithm')
+-->
+```Typescript
+    algo.transferTxId = await handleOrder(
+      providerInitializeComputeResults.algorithm,
+      resolvedAlgorithmDdo.services[0].datatokenAddress,
+      consumerAccount,
+      computeEnv.consumerAddress,
+      0
+    )
+    for (let i = 0; i < providerInitializeComputeResults.datasets.length; i++) {
+      assets[i].transferTxId = await handleOrder(
+        providerInitializeComputeResults.datasets[i],
+        dtAddressArray[i],
+        consumerAccount,
+        computeEnv.consumerAddress,
+        0
+      )
+    }
+    const computeJobs = await ProviderInstance.computeStart(
+      providerUrl,
+      web3,
+      consumerAccount,
+      computeEnv.id,
+      assets[0],
+      algo
+    )
+```
+<!--
+    assert(computeJobs, 'Cannot start compute job')
+-->
+Let's save the compute job it, we re going to use later
+```Typescript
+    computeJobId = computeJobs[0].jobId
+  
+```
+
+## 11. Check compute status and get download compute results URL
+  ### 11.1 Check compute status
+You can also add various delays so you see the various states of the compute job
+```Typescript
+    const jobStatus = await ProviderInstance.computeStatus(
+      providerUrl,
+      consumerAccount,
+      computeJobId,
+      DATASET_DDO.id
+    )
+```
+<!--
+    assert(jobStatus, 'Cannot retrieve compute status!')
+-->
+Now, let's see the current status of the previously started computer job
+```Typescript
+    console.log('Current status of the compute job: ', jobStatus)
+  
+```
+
+  ### 11.2 Get download compute results URL
+```Typescript
+    await sleep(10000)
+    const downloadURL = await ProviderInstance.getComputeResultUrl(
+      providerUrl,
+      web3,
+      consumerAccount,
+      computeJobId,
+      0
+    )
+```
+<!--
+    assert(downloadURL, 'Provider getComputeResultUrl failed!')
+-->
+Let's check the compute results url for the specified index
+```Typescript
+    console.log(`Compute results URL: ${downloadURL}`)
+  
+```
+
```

Comment:
please give me the full content of all code example files we instructed users to create here at the end.

Reply:
copied out all the code, ran `npx tsc` and got this:

<img width="714" alt="Screen Shot 2022-10-18 at 11 02 00" src="https://user-images.githubusercontent.com/90316/196400466-f042b7f9-a7b4-4185-9df4-72d18f07f880.png">

Reply:
and if the `tsconfig` would be valid, getting also bunch of TypeScript errors which prevent compiling:

<img width="849" alt="Screen Shot 2022-10-18 at 11 06 16" src="https://user-images.githubusercontent.com/90316/196401363-038227a2-7578-4d8b-a8c5-7a00f7d2b75c.png">

Reply:
tsconfig needs this line at the end:

```json
"exclude": [ "node_modules", "d" ]
```

Reply:
and then we get this, as we still compile to ES6. Either we compile to ES5 so it works in almost all Node.js versions, or we do `type: module` in the package.json but could lead to more issues then.

<img width="813" alt="Screen Shot 2022-10-18 at 11 48 52" src="https://user-images.githubusercontent.com/90316/196410178-796dcdec-9a1f-4c7a-b8b7-e7b25cc2d5fa.png">

Reply:
The best solution in this case I think is compiling to ES5, since also our tests are compiled to ES5 and are going to maintained